File: 2024/typescript/easy/threeConsecutiveOdds.ts
function threeConsecutiveOdds(arr: number[]): boolean {
  let count = 0;

  for (let i = 0; i < arr.length; i++) {
    if (count === 3) return true;

    if (arr[i] % 2 !== 0) {
      count++;
    } else {
      count = 0;
    }
  }

  return count === 3 ? true : false;
}

File: 2024/go/easy/threeConsecutiveOdds.go
package easy

//lint:ignore U1000 Ignore unused function
func threeConsecutiveOdds(arr []int) bool {
	count := 0

	for _, v := range arr {
		if count == 3 {
			return true
		}

		if v%2 != 0 {
			count++
		} else {
			count = 0
		}
	}

	return count == 3
}
